test/functional/ui/decorations_spec.lua

```diff
@@ -1779,6 +1779,25 @@ l5
     ]]}
   end)
 
+  it('works with priority #19716', function()
```

Comment:
Maybe add a second test with `signcolumn=auto` to test correct truncation (as this was a separate fix)?